Local ARS (Adaptive Routing and Switching)

VladimirKuk opened the pull request:
Added support for local ARS (Adaptive Routing and Switching).

HLD: https://github.com/sonic-net/SONiC/pull/1958

VladimirKuk committed:
Move to new branch

Signed-off-by: Vladimir Kuk <vkuk@marvell.com>

VladimirKuk committed:
Resolving merge conflict

Signed-off-by: Vladimir Kuk <vkuk@marvell.com>

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

prsunny commented on common/schema.h:
```diff
@@ -259,34 +263,35 @@ namespace swss {
-#define PLUGIN_TABLE                    "PLUGIN_TABLE"
-#define LUA_PLUGIN_TYPE                 "LUA_PLUGIN_TYPE"
-#define SAI_OBJECT_TYPE                 "SAI_OBJECT_TYPE"
+#define PFC_WD_POLL_MSECS                 100
```
Please revert the changes not applicable to this feature

apannerselva replied:
resolved

mssonicbld commented:
/azp run

prsunny commented on common/schema.h:
```diff
@@ -54,7 +54,6 @@ namespace swss {
 #define APP_SWITCH_TABLE_NAME             "SWITCH_TABLE"
 #define APP_NEXTHOP_GROUP_TABLE_NAME      "NEXTHOP_GROUP_TABLE"
 #define APP_CLASS_BASED_NEXT_HOP_GROUP_TABLE_NAME "CLASS_BASED_NEXT_HOP_GROUP_TABLE"
-#define APP_EVPN_SPLIT_HORIZON_TABLE_NAME    "EVPN_SPLIT_HORIZON_TABLE"
```
Why remove existing definition?

apannerselva commented:
/azpw run Azure.sonic-swss-common (Test vstest)

rck-innovium commented:
hi @prsunny We have addressed all the open comments. Can you please review and merge.

rustiqly commented:
/azp run

apannerselva commented:
/azpw run Azure.sonic-buildimage

apannerselva commented:
/AzurePipelines run Azure.sonic-buildimage

mssonicbld commented:
/azp run

VladimirKuk committed:
Local ARS (Adaptive Routing and Switching)

Added support for local ARS (Adaptive Routing and Switching).

HLD: https://github.com/sonic-net/SONiC/pull/1958

Signed-off-by: VladimirKuk <31180446+VladimirKuk@users.noreply.github.com>
